File: vol2/ch17/LeetCode74.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m,n=len(matrix),len(matrix[0])
        i,j=0,n-1
        while True:
            if target==matrix[i][j]:return True
            elif target<matrix[i][j]:j-=1
            else: i+=1
            if i>=m  or j<0:return False
        return False 

File: vol2/ch17/LeetCode74.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m=matrix.size(),n=matrix[0].size();
        int i=0,j=n-1;
        while(true) {
            if(target==matrix[i][j]) return true;
            else if(target<matrix[i][j]) j--;
            else i++;
            if(i>=m || j<0) return false;
        }
        return false;
    }
};

